### PR TITLE
[#63] [Chore] Setup BuildKonfig

### LINKED ISSUE #63: [Chore] Setup BuildKonfig
## Why

Passing values from Android/iOS or any other platform code should work, but it's a hassle. So, setting up the KMM to read values from properties and add those into BuildConfig is the better way

## Who Benefits?

Developer

### PR DESCRIPTION
**Pull request by markgravity**
https://github.com/markgravity/kmm-ic/issues/63

## What happened 👀

Setup BuildKonfig

## Insight 📝

Separated into a small story from #15, so it's easier for reviewers to review

## Proof Of Work 📹

CleanShot 2023-04-19 at 09 32 06

### CODE REVIEW

File: .github/workflows/deploy_production_appstore.yml [hunk 1]
```diff
@@ -51,6 +51,13 @@ jobs:
         touch KMMIC/Configurations/Plists/GoogleService/Production/GoogleService-Info.plist
         echo $IOS_GOOGLE_SERVICE_INFO_PRODUCTION | base64 --decode > ./KMMIC/Configurations/Plists/GoogleService/Production/GoogleService-Info.plist
 
+    - name: Setup Konfig Properties
+      env:
+       BUILD_KONFIG_PROPERTIES: ${{ secrets.BUILD_KONFIG_PROPERTIES }}
+      run: |
+       touch ../buildKonfig.properties
+       echo $BUILD_KONFIG_PROPERTIES | base64 --decode > ../buildKonfig.properties
```
Comment (phongvhd93) [inline .github/workflows/deploy_production_appstore.yml:59]: Did you add `buildKonfig.properties` to gitignore?

Comment (markgravity) [reply .github/workflows/deploy_production_appstore.yml:59]: @phongvhd93 I already did it [here](https://github.com/markgravity/kmm-ic/blob/99945a33e341dbb801ff394748f06686a1d05da9/.gitignore#L139)

File: build.gradle.kts [hunk 2]
```diff
@@ -1,3 +1,15 @@
+buildscript {
+    repositories {
+        gradlePluginPortal()
+        google()
+        mavenCentral()
+    }
+    dependencies {
+        classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:1.6.21")
```
Comment (phongvhd93) [inline build.gradle.kts:8]: How about setting up a `buildSrc` module to manage these dependencies?

Comment (markgravity) [reply build.gradle.kts:8]: I missed this one, let me implement it on another chore task after #67 is merged 🙏

**Review by phongvhd93 — APPROVED** [review]

**Review by suho — APPROVED** [review]

**Comment by suho** [comment]
@markgravity Don't forget to check CI statues

**Pull request merged**